charmcraft/commands/build.py

```diff
@@ -79,6 +81,21 @@ def polite_exec(cmd):
     return retcode
 
 
+def relativise(src, dst):
```

Comment:
Could we just use `os.path.relpath()` here? Something like `os.path.relpath(dst, src.parent)` should return the same thing.

Reply:
`os.path.relpath` falls flat on its face for most of these:

| `src` | `dst` | `relativise(src, dst)` | `os.path.relpath(dst, src.parent)` |
| --- | --- | -- | -- |
| `/tmp/foo/bar/src.txt` | `/tmp/foo/bar/dst.txt` | `dst.txt` | `dst.txt` |
| `/tmp/foo/bar/baz/src.txt` | `/tmp/foo/dst.txt` | `../../dst.txt` | `dst.txt` |
| `/tmp/foo/src.txt` | `/tmp/foo/bar/baz/dst.txt` | `bar/baz/dst.txt` | `dst.txt` |
| `/tmp/foo/bar/src.txt` | `/tmp/foo/baz/dst.txt` | `../baz/dst.txt` | `dst.txt` |
| `/tmp/foo/bar1/bar2/src.txt` | `/tmp/foo/baz1/baz2/baz3/dst.txt` | `../../baz1/baz2/baz3/dst.txt` | `dst.txt` |

Reply:
This is interesting to me. I didn't directly run the test cases, but I locally tested these 4 and they return the same result using `os.path.relpath`

e.g. 
```
>>> src = pathlib.Path("/tmp/foo/bar1/bar2/src.txt")
>>> dst = pathlib.Path("/tmp/foo/baz1/baz2/baz3/dst.txt")
>>> os.path.relpath(dst, src.parent)
'../../baz1/baz2/baz3/dst.txt'
```

Reply:
I don't know what I did to get those results now. I can reproduce what you're doing there and get what you got. Sadly the script I used to create the table was in `/tmp` and I had to reboot :-(
But hey, this means less code \o/